content/research/market-health/posts/2026-06-12-celr-repeated-lot-clustering/scripts/fetch_and_analyze.py

```
@@ -0,0 +1,321 @@
+def metrics(symbol, rows, zip_bytes):
+    qty_counts = Counter(row["qty_text"] for row in rows)
+    top = qty_counts.most_common(5)
+    timestamps = [row["timestamp"] for row in rows]
+    prices = [row["price"] for row in rows]
+    qty_texts = [row["qty_text"] for row in rows]
+    seconds = Counter((timestamp // 1000) % 60 for timestamp in timestamps)
+    expected = len(rows) / 60
+    chi_square = sum((seconds.get(second, 0) - expected) ** 2 / expected for second in range(60))
+    gaps = [
+        (current - previous) / 1000
+        for previous, current in zip(timestamps, timestamps[1:])
```

Ruff (0.15.15)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)